app/src/app/session/login/login.component.html

```diff
@@ -15,7 +15,9 @@
     <p *ngSwitchCase="'Local'" i18n>Internal error: {{errorMsg}}</p>
     <p *ngSwitchCase="'Server'" i18n>Server error: {{errorMsg}}</p>
   </div>
-  <div class="formactions">
+
+  <div class="formactions login">
```

Comment:
This does not correspond to what you claimed to do: `formactions login` is very different from `formactions-login`. In your code, the div element belongs to two classes `formactions` and `login`. This is not a wrong solution, but the name `login` seems a bit too general to me for this element. It could be a good additional class for the form though, in which case you could use a selector like `form.login .formactions`. Another solution is to just rename this class to something like `login-action`.

Reply:
i will rename it to login-action